webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactStorage.scala

```diff
@@ -696,7 +708,16 @@ class LdapNodeFactStorage(
                           _ <- ZIO.foreach(newSoft)(x => con.save(inventoryMapper.entryFromSoftware(x)))
                         } yield Some(newSoft.map(_.id) ++ merged.alreadySavedSoftware.map(_.id))
                     }
-      optOld     <- getNodeFact(nodeFact.id, nodeFact.rudderSettings.status, SelectFacts.noSoftware).map(
+      // here we can be more selective: if no inventory is needed, don't retrieve it.
+      // But as seen as we need a bit of inventory, we need to find it all, since repo delete missing bit -even
+      // for "onlySoftware", because we save software in inventory node, too.
+      getAttrs    = if (attrs == SelectFacts.none) {
+                      SelectFacts.none
+                    } else {
+                      if (needsSoftware(attrs)) SelectFacts.all
+                      else SelectFacts.noSoftware
+                    }
+      optOld     <- getNodeFact(nodeFact.id, nodeFact.rudderSettings.status, getAttrs).map(
```

Comment:
Is it fetching two times the software when we need them ?

Reply:
hum, perhaps - I didn't see it in tests, but nothing from the code let me think it does not

Reply:
it looks like indeed, we don't have to fetch software, and the before-pull request was right on that